smart_open/daos.py

```diff
@@ -0,0 +1,339 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2022 Hewlett Packard Enterprise Development LP.
+#
+# This code is distributed under the terms and conditions
+# from the MIT License (MIT).
+#
+
+'''
+Implements reading and writing to/from DAOS
+
+The main entry point is the :func:`~smart_open.daos.open` function.
+
+Uses the pyDAOS KV store under the covers.
+
+'''
+
+import io
+import logging
+import subprocess
+import urllib.parse
+from smart_open import utils
+import smart_open.utils
+import time
+
+try:
+    import sys
+    sys.path.append("/opt/daos/lib64/python3.6/site-packages/")
```

Comment:
This seems non-ideal. As I understand it python3.7 is the minimum supported for smart open https://github.com/RaRe-Technologies/smart_open/pull/688 ?

Reply:
DAOS supports python with pyDAOS(https://www.intel.com/content/www/us/en/developer/articles/case-study/unlock-the-power-of-daos-in-python-with-pydaos.html) using python3.6. This is packaged separately. Not dependent on the smart_open supported version.

Reply:
Why are you even touching sys.path inside the Python source?

Reply:
This is  a requirement when DAOS is not installed from repository packages. Will remove this.